boards/shields/nrf7002ek/nrf7002ek_coex.overlay

```diff
@@ -0,0 +1,15 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ /* Include the base overlay file for the nRF7002EK shield */
+ #include "nrf7002ek.overlay"
+
+&nrf70 {
+	status0-gpios = <&arduino_header 8 GPIO_ACTIVE_HIGH>;       /* D2 */
+	req-gpios = <&arduino_header 9 GPIO_ACTIVE_HIGH>;   /* D3 */
+	grant-gpios = <&arduino_header 10 (GPIO_PULL_DOWN | GPIO_ACTIVE_LOW)>;    /* D4 */
+	swctrl1-gpios = <&arduino_header 12 GPIO_ACTIVE_HIGH>;  /* D6 */
+	srrf-switch-gpios = <&arduino_header 14 GPIO_ACTIVE_HIGH>;  /* D8 */
```

Comment:
You should move this to the `nrf7002ek.overlay` alongside `/* D0 */` :) The hardware layout does not change whether the app/driver uses these pins or not. Then you can simply remove `nrf7002ek_coex.overlay`.

A driver can simply check if the pins are connected using `DT_INST_HAS_PROP(0, status0_gpios)` and similar to check if all required pins for coex exist if coex is enabled in the driver :)

Reply:
> You should move this to the nrf7002ek.overlay alongside /* D0 */ :) The hardware layout does not change whether the app/driver uses these pins or not. Then you can simply remove nrf7002ek_coex.overlay.

As these GPIOs are dependent on host support (not just nRF70) how would we make them conditional if we move them `nrf7002-spi` node? E.g., on nRF5340 host these are applicable and could be enabled/disabled, but on nRF91 series these aren't applicable at all, so, these GPIOs can be used for other needs.

> A driver can simply check if the pins are connected using DT_INST_HAS_PROP(0, status0_gpios) and similar to check if all required pins for coex exist if coex is enabled in the driver :)

It does that already.

Reply:
When the shield is put onto any board with an arduino connector, the coex pins on the nRF7002 will be physically connected to the hosts GPIOs. The COEX pins can not be reused on an nRF91, as they are connected to the nRF7002...

Reply:
Let's run this through an example: 

> status0-gpios = <&arduino_header 8 GPIO_ACTIVE_HIGH>;       /* D2 */
This maps to ` <8 0 &gpio1 4 0>,	/* D2 */`.

For nRF91 series host, the `gpio1.4` can be removed from the `arduino_header`  (as that physical pin is unused) and it can be used for other peripheral. So, it would save a total of 5GPIOs without COEX, this was the primary motivation to have it separate (either overlay or a shield)

Am I missing anything?

Reply:
I'm expecting you mean "for nRF53 series host", the physical pin `gpio1.4` may be "unused" but it is physically connected to `status0-gpios`, which is an output from the nRF7002, when the shield is on the board.

If the user really wishes to use gpio1.4 while the shield is on the board, the user must physically alter the board or shield to cut the physical connection, at which point the user can add
```
&nrf70 {
        /delete-property/ &status0-gpios;
};
```
to their overlay to indicate the connection has been removed, and can then use `gpio1.4` for something else... The devicetree describe the physical connections

Reply:
> I'm expecting you mean "for nRF53 series host", t

No, these can be used in nRF53, for nRF91 these aren't applicable, so, was trying to give that example of re-use.

> If the user really wishes to use gpio1.4 while the shield is on the board, the user must physically alter the board or shield to cut the physical connection,

True, we had this request from a customer, but I don't have the details, they might be probably using wires to connect pins, so, not needed to modify hardware.

> at which point the user can add
&nrf70 {
        /delete-property/ &status0-gpios;
};
to their overlay to indicate the connection has been removed, and can then use gpio1.4 for something else

True, this works too, and is simpler I guess.

> A driver can simply check if the pins are connected using DT_INST_HAS_PROP(0, status0_gpios)

If we move these pins to main nRF70 node, wouldn't this be always true? I don't think from DTS we would know if a pin is physically connected or not, do we?

Reply:
ok, so, the conclusion is:

* Remove the `nrf7002_coex` shield variant
* Merge coex pins in to main `nrf7002-spi` node
* Convert `NRF70_SR_COEX` and `NRF70_SR_COEX_RF_SWITCH` to user configurable ones instead of DTS based

@jfischer-no hope you are okay with this proposal.

Reply:
> * Convert `NRF70_SR_COEX` and `NRF70_SR_COEX_RF_SWITCH` to user configurable ones instead of DTS based

You can use shield variants to set these Kconfig options in board/shields/nrf7002ek/Kconfig.defconfig.

Reply:
With this review comment, I have now included co-ex pins uncondtionally (and removed coex shield variant), so, cannot rely on existence of pins anymore, so, moved these options for user to enable COEX.